modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/model/listener/test/DDMStructureModelListenerTest.java

```diff
@@ -300,6 +311,63 @@ private void _assertDDMFormFieldValuesMap(
 		}
 	}
 
+	private void _assertUpdateDataDefinitionStructureKeyChanged()
+		throws Exception {
+
+		JournalArticle journalArticle = _addJournalArticle();
+
+		String modifiedDataDefinitionKey =
+			_dataDefinition.getDataDefinitionKey() + "-MODIFIED";
+
+		_updateDataDefinition(
+			modifiedDataDefinitionKey, "dependencies/data_definition.json");
+
+		journalArticle = _journalArticleLocalService.getJournalArticle(
+			journalArticle.getId());
+
+		_assertDDMFormFieldValuesMap(
+			_expectedFieldValuesMap, journalArticle.getDDMFormValues());
+
+		Indexer<JournalArticle> indexer =
+			IndexerRegistryUtil.nullSafeGetIndexer(JournalArticle.class);
+
+		Document document = indexer.getDocument(journalArticle);
+
+		Assert.assertEquals(
+			modifiedDataDefinitionKey, document.get("ddmStructureKey"));
+	}
+
+	private void _assertUpdateStructureKey() throws Exception {
```

Comment:
Same as before. Additionally, the method name should tell what it is asserting. Something like `_assertStructureKeyDidntChange` would be enough.